gempython/tests/amc13ReadOut.py

```diff
@@ -0,0 +1,198 @@
+    parserTrigGroup.add_argument("--t3trig", action="store_true", dest="t3trig",
+            help="Set up for using AMC13 T3 trigger input")
+    parserTrigGroup.add_argument("--internal", action="store_true", dest="internal",
+            help="Configure the amc13 to generate local triggers")
+
+    args = parser.parse_args()
+    options = vars(args)
+    
+    # Get the amc13
+    import uhal
+    if args.debug:
+        uhal.setLogLevelTo(uhal.LogLevel.INFO)
+    else:
+        uhal.setLogLevelTo(uhal.LogLevel.ERROR)
+    from gempython.tools.xdaq.amc13manager import AMC13manager
+    amc13base  = "gem.shelf%02d.amc13"%(args.shelf)
+    amc13board = AMC13manager()
+    amc13board.connect(amc13base,args.debug)
+    
+    # Get the slots
+    from gempython.tools.amc_user_functions_uhal import *
+    dict_amcs = {}
+    slotList = args.slotList.split(',')
+    for slotRange in slotList:
+        if "-" in slotRange:
+            slotsInRange = [int(slot) for slot in slotRange.split("-")]
+            for slot in range(min(slotsInRange),max(slotsInRange)+1):
+                dict_amcs[slot] = getAMCObject(slot,shelf=args.shelf,debug=args.debug)
+        else:
+            dict_amcs[slotRange] = getAMCObject(int(slotRange),shelf=args.shelf,debug=args.debug)
+
+    # Define the parallel processes
+    from multiprocessing import Process
+    dataTaking = Process(target=startDataTaking, args=(amc13board,args.filename))
+    halt = Process(target=stopDataTaking, args=(amc13board,args.acquireTime))
+
+    # Stop Triggers
+    for slot,amc in dict_amcs.iteritems():
+        blockL1A(amc)
+    amc13board.device.enableLocalL1A(False)
+
+    # Setup for data-taking and get L1A counters
+    amc13board.reset()
+    amc13board.configureInputs(args.slotList)
+    trigCount = readAllL1ACounters(amc13board, dict_amcs, "INITIAL")
+    #amc13board.device.monBufBackPressEnable(False)
+
+    # Update DAQ Input Enable Mask for each AMC and turn on ZS if requested
+    for slot,amc in dict_amcs.iteritems():
+        # Zero suppression
+        if args.zeroSup:
+            writeRegister(amc,"GEM_AMC.DAQ.CONTROL.ZERO_SUPPRESSION_EN",0x1)
+        else:
+            writeRegister(amc,"GEM_AMC.DAQ.CONTROL.ZERO_SUPPRESSION_EN",0x0)
+
+        # DAQ Link Enable and input mask
+        if args.ohMask < 0:
+            scaRdy = readRegister(amc,"GEM_AMC.SLOW_CONTROL.SCA.STATUS.READY")
+            enableDAQLink(amc,scaRdy)
+        else:
+            enableDAQLink(amc,args.ohMask)
+
+    # Enable triggers to amc's
+    for slot,amc in dict_amcs.iteritems():
+        enableL1A(amc)
+    amc13board.device.enableLocalL1A(True)
+    amc13board.configureTrigger(ena=args.internal,t3trig=args.t3trig)
+    if args.internal:
+        amc13board.configureCalPulse()
+
+    # Take data
+    import sys
+    try:
+        #dataTaking.start()
+        dataTaking.run()
+        #halt.start()
+        halt.run()
+        halt.join()
+        print("Data Taking has completed")
```

Comment:
@mexanick I am not sure how to get `halt` to stop ` dataTaking` process.  Somehow I think a `multiprocessing.queue` object needs to be taken as an argument instead of an instance of the class `AMC13manager`.  But what should be put into the queue I am unsure...perhaps start and stop methods of `AMC13manager` should be taking a queue as argument and the queue takes some timeout value? 

I have played around but haven't really gotten anywhere.

Right now a `KeyboardInterrupt` will gracefully close the program.  And if not suitable other solution can be found I would keep with this and drop the `acquireTime` argument altogether (along with `multiprocessing` objects).